Formula/rpm.rb

```diff
@@ -27,51 +13,40 @@ class Rpm < Formula
     sha256 "3a276b6a3f7273f8f88effbb057e8d65fc53fa9e812054951bfdd5ce618f54ac" => :yosemite
   end
 
-  depends_on "rpm2cpio" => :build
+  depends_on "beecrypt"
   depends_on "berkeley-db"
   depends_on "libmagic"
+  depends_on "libarchive"
   depends_on "popt"
-  depends_on "libtasn1"
+  depends_on "gnupg"
   depends_on "gettext"
+  depends_on "lua"
   depends_on "xz"
-  depends_on "ossp-uuid"
+  depends_on "zstd"
+  depends_on "pkg-config" => :run
 
   def install
     # only rpm should go into HOMEBREW_CELLAR, not rpms built
-    inreplace "macros/macros.in", "@prefix@", HOMEBREW_PREFIX
+    inreplace "macros.in", "@prefix@", HOMEBREW_PREFIX
+    inreplace "platform.in", "@prefix@", HOMEBREW_PREFIX
     args = %W[
       --prefix=#{prefix}
       --localstatedir=#{var}
-      --with-path-cfg=#{etc}/rpm
+      --sharedstatedir=#{var}/lib
+      --sysconfdir=#{etc}
       --with-path-magic=#{HOMEBREW_PREFIX}/share/misc/magic
-      --with-path-sources=#{var}/lib/rpmbuild
       --with-libiconv-prefix=/usr
-      --disable-openmp
-      --disable-nls
+      --enable-nls
       --disable-dependency-tracking
-      --with-db=external
-      --with-sqlite=external
-      --with-file=external
-      --with-popt=external
-      --with-beecrypt=internal
-      --with-libtasn1=external
-      --with-neon=internal
-      --with-uuid=external
-      --with-pcre=internal
-      --with-lua=internal
-      --with-syck=internal
+      --disable-plugins
+      --with-external-db
+      --with-crypto=beecrypt
```

Comment:
are there other choices?

Reply:
The other choices are NSS and OpenSSL (1.0.2+/1.1+ only). I wanted to avoid potential license conflicts with OpenSSL (OpenSSL license / GPLv2+ license), and system OpenSSL is too old anyway. NSS is the backend used in Fedora, Mageia, and a few others, while BeeCrypt is the backend used in the SUSE family of distributions.

Reply:
@DomT4 any thoughts here?

Reply:
I thoroughly hate bundled crypto libraries, not least ones that have been booted out of Debian for being too inactive upstream. Debian use `nss`. We don't tend to care much about inflicting the OpenSSL license on everything else in Homebrew & I don't think we should care much here, to be honest, but I'd take anything that is not a bundled crypto library.

Reply:
Well, there's no bundled crypto library here. 👍

Reply:
Ah, I see you switched it to the formula. Doesn't really change the fact it seems to be dead in terms of upstream releases & Debian have booted it out of newer distros because of that? If nobody is doing regular security work on it, I don't want formulae using it, personally.

Reply:
Frankly now I've been reminded `beecrypt` is offered as a Homebrew formula I'm kinda bothered it exists at all 🙈.

Reply:
Probably should just use `depends_on "openssl"` until we come up with some non-license reason not to.

Reply:
I can switch it to OpenSSL if you'd prefer that?

Reply:
@DomT4 I wonder what these folks are doing with it
```
Josephs-MacBook-Pro:Formula joe$ brew formula-analytics beecrypt
install events in the last 30 days for beecrypt
======================================================================================
1 | beecrypt                                                            | 86 | 100.00%
======================================================================================
Total                                                                   | 86 |    100%
======================================================================================
```
It seems like just enough people to keep it and just few enough not to care.

Reply:
@Conan-Kudo yes openssl means 1.0.2l currently. We're always on the latest.

Reply:
Alrighty then...

Reply:
> I can switch it to OpenSSL if you'd prefer that? But I'd need to be sure it's 1.0.2+...

Homebrew hasn't shipped anything older for quite some time.

> I wonder what these folks are doing with it

Googling cryptographic libraries & stumbling upon suggestions from 2011? 🙈. Honestly I have no idea.

`beecrypt` _could_ still be perfectly safe, usable software but given there hasn't been an upstream release in 8 years now that's a heck of a leap of faith to take with a cryptographic library. Even the relatively simple ones have had some nasty vulnerabilities. That Debian have booted it out of existence on newer distros certainly doesn't fill me with confidence.

Reply:
I'm having trouble getting it to link to Homebrew OpenSSL. Are there any special tricks I should do to make that work? This is the logic for OpenSSL detection: https://github.com/rpm-software-management/rpm/blob/master/configure.ac#L315-L393

Reply:
Can you provide the failure? It doesn’t look like it’s doing anything too weird, although am on a phone so could be missing something subtle.

Reply:
```
checking openssl/evp.h usability... yes
checking openssl/evp.h presence... yes
checking for openssl/evp.h... yes
checking openssl/rsa.h usability... yes
checking openssl/rsa.h presence... yes
checking for openssl/rsa.h... yes
checking openssl/dsa.h usability... yes
checking openssl/dsa.h presence... yes
checking for openssl/dsa.h... yes
checking for EVP_DigestInit_ex in -lcrypto... yes
checking for EVP_MD_CTX_new in -lcrypto... no
checking for EVP_MD_CTX_create in -lcrypt... no
configure: error: required OpenSSL library 'libcrypto' missing or too old
```

Reply:
I think OpenSSL is best choice

Reply:
@Conan-Kudo I think you may have inadvertently introduced an OpenSSL 1.1 minimum.

Reply:
@ilovezfs I just checked, and I think you're right. The OpenSSL 1.1 formula works.

Reply:
Heh, I wonder if upstream know about this or whether they slipped into 1.0.2 incompatibility accidentally. NSS isn’t a bad alternative if necessary.

Reply:
Nah, OpenSSL 1.1 is a perfectly fine thing to use. Besides, it's the future! 💯

Reply:
Was more intended for ILZ. OpenSSL 1.1 in Homebrew has become a complicated subject.

Reply:
I think the `checking for EVP_MD_CTX_create in -lcrypt... no` is a bug.

Reply:
Yeah it should `lcrypto` not `lcrypt`.

Reply:
```
iMac-TMP:openssl joe$ brew linkage rpm
System libraries:
  /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
  /usr/lib/libSystem.B.dylib
  /usr/lib/libbz2.1.0.dylib
  /usr/lib/libiconv.2.dylib
  /usr/lib/libz.1.dylib
Homebrew libraries:
  /usr/local/opt/berkeley-db/lib/libdb-6.2.dylib (berkeley-db)
  /usr/local/opt/gettext/lib/libintl.8.dylib (gettext)
  /usr/local/opt/libarchive/lib/libarchive.13.dylib (libarchive)
  /usr/local/opt/libmagic/lib/libmagic.1.dylib (libmagic)
  /usr/local/opt/lua/lib/liblua.5.2.dylib (lua)
  /usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib (openssl)
  /usr/local/opt/popt/lib/libpopt.0.dylib (popt)
  /usr/local/Cellar/rpm/4.14.0_1/lib/librpm.8.dylib (rpm)
  /usr/local/Cellar/rpm/4.14.0_1/lib/librpmbuild.8.dylib (rpm)
  /usr/local/Cellar/rpm/4.14.0_1/lib/librpmio.8.dylib (rpm)
  /usr/local/Cellar/rpm/4.14.0_1/lib/librpmsign.8.dylib (rpm)
  /usr/local/opt/xz/lib/liblzma.5.dylib (xz)
  /usr/local/opt/zstd/lib/libzstd.1.dylib (zstd)
```

Reply:
Oh, wow. Did not spot that on the phone-sized screen. In some way that’s actually hilarious.

Reply:
Oh geez. This... is sad. I'll send a patch upstream asap.

Reply:
https://github.com/rpm-software-management/rpm/pull/335

Reply:
Occam's razor advocates will be thrilled with this troubleshooting ultimately leading to one tiny tiny typo.